### PR TITLE
Support ghcr.io

### PR DESCRIPTION
**Pull request by zeroalphat**
zeroalphat <taichi-takemura@cybozu.co.jp>

### CODE REVIEW

File: .github/actions/build_push/action.yaml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Extract targets
+      shell: bash
+      run: |
+        dir=${{ inputs.dir }}
+        if [ "$dir" = "" ]; then dir=${{inputs.container-image }} ; fi
```
Comment (morimoto-cybozu) [inline .github/actions/build_push/action.yaml:25]: for readability

```suggestion
        if [ "$dir" = "" ]; then dir=${{ inputs.container-image }} ; fi
```

File: .github/actions/build_push/action.yaml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+              fi
+            fi
+        fi
+      env:
+        GITHUB_TOKEN: ${{ inputs.github_token }}
+    - name: Validate consistency between BRANCH and TAG
+      shell: bash
+      run: |
+        dir=${{ inputs.dir }}
+        if [ "$dir" = "" ]; then dir=${{inputs.container-image }} ; fi
```
Comment (morimoto-cybozu) [inline .github/actions/build_push/action.yaml:49]: ditto

```suggestion
        if [ "$dir" = "" ]; then dir=${{ inputs.container-image }} ; fi
```

**Review by morimoto-cybozu — CHANGES_REQUESTED** [review]

File: .github/actions/setup/action.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+      with:
+        platforms: linux/adm64,linux/arm64/v8
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        version: v0.9.1
```
[the remark this answers is not in the document]
Comment (zeroalphat) [reply .github/actions/setup/action.yaml]: The latest version of container-tag-exists is now able to determine tags correctly.
Fixing the version is no longer necessary.
https://github.com/Hsn723/container-tag-exists/issues/20
cbdcc45b77af52128b17f16be9b1897c4afd91ca

File: .github/actions/build_push/action.yaml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    - name: Extract targets
+      shell: bash
+      run: |
+        dir=${{ inputs.dir }}
+        if [ "$dir" = "" ]; then dir=${{inputs.container-image }} ; fi
+        image=${{ inputs.container-image }}
+        TAG=$(cat ${dir}/TAG)
+        c=$(container-tag-exists ghcr.io/cybozu/$image $TAG 2>&1)
+        # The stderr should be either "" or "found".
+        # If the stderr was an error message other than "found", this step must have stopped due to the shell option "-e".
```
[the remark this answers is not in the document]
Comment (zeroalphat) [reply .github/actions/build_push/action.yaml]: Fix at b7e71cf29604a53bc9e99dc21fbd93d9b7769153.

File: .github/actions/build_push/action.yaml (hunk 2 above)
Comment (zeroalphat) [reply .github/actions/build_push/action.yaml]: Fix at dae0f70d53a7158fed7fea616c7995f42630b789.

File: .github/actions/build_push/action.yaml (hunk 1 above)
Comment (zeroalphat) [reply .github/actions/build_push/action.yaml]: Fix at dae0f70d53a7158fed7fea616c7995f42630b789.

File: .github/actions/build_push/action.yaml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        if [ "$dir" = "" ]; then dir=${{inputs.container-image }} ; fi
+        if [ -e "$dir/NO_TAG_BRANCH_CONSISTENCY" ]; then exit 0; fi
+        ./tag_branch_consistency $dir
+    - name: Build and push
+      if: env.BUILDS_TAG != null && github.event_name != 'pull_request'
+      uses: docker/build-push-action@v4
+      with:
+        context: ${{ inputs.dir }}
+        platforms: linux/amd64,linux/arm64
+        push: false
```
[the remark this answers is not in the document]
Comment (zeroalphat) [reply .github/actions/build_push/action.yaml]: I was thinking of modifying push to true in the next PR.
I also think it is better if the build is executed when the PR is created, so I changed the conditions under which the build is executed.
c43f95e5396b3ca1896dac859728b60fe1979f0c

**Review by morimoto-cybozu — APPROVED** [review]
LGTM

**Review by morimoto-cybozu — APPROVED** [review]

**Pull request merged**